### PR TITLE
ex5 - classes exercise

### PR DESCRIPTION
**Pull request by IsraelCohen54**
ex5 - classes exercise
I would like any feedback, thank!

### CODE REVIEW

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.hibob.ex5_classes_exercise
+
+data class Location(val street: String, val city: String, val county: String, val zipCode: Int?=null, val postCode: Int?=null)
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt]: should all the locations hold properties only specific location has (in us there is zip but no post and vice versa)?

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    init {
+        // Ensure that at least one of zipCode or postCode is provided
+        require(zipCode != null || postCode != null) {
+            "Either zipCode or postCode must be provided, but not both."
+        }
+    }
+}
+
+open class Meeting(val name: String, val locationDetails: List<Location>) {
+    private lateinit var participant: MutableList<Participant>
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt:16]: nice

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    fun addParticipant(participant: Participant) {
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt:18]: very good!

**Review by liorsiv973 — APPROVED** [review]
very good!! see few comments

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt (hunk 1 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt]: That does make sense, thanks for the feedback! 
If I may ask, I would like to know what would be the best approach in your opinion? Should I concat and write only code as field, then by country type I would know which type of code is it with
`when(country) ... is US -> zipcode` etc?
or something else?
Best regards

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt (hunk 3 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt:18]: thank you very much!

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt (hunk 1 above)
Comment (liorsiv973) [reply src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt]: I would make location as interface and then US/UK as implementation

Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt]: Like this (updated file)?

File: src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.hibob.ex5_classes_exercise
+
+interface Location{}
+
+data class Us(val zipCode: Int) : Location {}
+
+data class Uk(val postalCode: Int) : Location {}
+
+data class LocationDetails(val street: String, val city: String, val county: String, val loc:Location) {}
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt]: why not adding all of this to the location interface?

Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/ex5_classes_exercise/Meeting.kt]: updated, thanks!

**Pull request merged**